interfaces/builtin/microstack_support.go

```diff
@@ -0,0 +1,256 @@
+/sys/devices/platform/host*/session*/connection*/iscsi_connection/connection*/** rw,
+/sys/devices/platform/host*/session*/iscsi_session/session*/** rw,
+/sys/devices/platform/host*/session*/target*/** rw,
+/sys/devices/platform/host*/iscsi_host/host*/** rw,
+
+# While the block-devices interface allows rw access, Libvirt also needs to be able to lock those.
+/dev/sd{,[a-h]}[a-z] rwk,
+/dev/sdi[a-v] rwk,
+# os-brick needs access to those when detaching a scsi device from an instance.
+/sys/block/sd{,[a-h]}[a-z]/device/delete rw,
+/sys/block/sdi[a-v]/device/delete rw,
+
+# Used by open-iscsi to avoid being killed by the OOM killer.
+owner @{PROC}/@{pid}/oom_score_adj rw,
+
+
+# Allow running utility processes under the specialized AppArmor profiles.
+# These profiles will prevent utility processes escaping confinement.
+capability mac_admin,
+
+# MicroStack services such as libvirt use a server/client design where
+# unix sockets are used for IPC.
+capability chown,
+
+# Required by Nova.
+capability dac_override,
+capability dac_read_search,
+capability fowner,
+
+# Used by libvirt to alter process capabilities via prctl.
+capability setpcap,
+# Used by libvirt to create device special files.
+capability mknod,
+
+# Allow libvirt to apply policy to spawned VM processes.
+change_profile -> libvirt-[0-9a-f]*-[0-9a-f]*-[0-9a-f]*-[0-9a-f]*-[0-9a-f]*,
+
+# Allow sending signals to the spawned VM processes.
+signal (read, send) peer=libvirt-*,
+
+# Allow reading certain proc entries, see ptrace(2) "Ptrace access mode checking".
+# For ourselves:
+ptrace (read, trace) peer=@{profile_name},
+# For VM processes libvirt spawns:
+ptrace (read, trace) peer=libvirt-*,
+
+# Used by neutron-ovn-agent.
+unmount /run/netns/ovnmeta-*,
+`
+
+const microStackSupportConnectedPlugSecComp = `
+# Description: allow MicroStack to operate by allowing the necessary system calls to be used by various services.
+# (libvirt, qemu, qemu-img, Nova, Neutron, Keystone, Glance, Cinder)
+
+# Note that this profile necessarily contains the union of all the syscalls each of the
+# utilities requires. We rely on MicroStack to generate specific AppArmor profiles
+# for each child process, to further restrict their abilities.
+mknod - |S_IFBLK -
+mknodat - - |S_IFBLK -
+`
+
+var microStackConnectedPlugUDev = []string{
+	`KERNEL=="vhost-net"`,
+	`KERNEL=="vhost-scsi"`,
+	`KERNEL=="vhost-vsock"`,
+	`SUBSYSTEM=="block", KERNEL=="nbd[0-9]*"`,
+	`SUBSYSTEM=="misc", KERNEL=="vfio"`,
+	`SUBSYSTEM=="vfio", KERNEL=="[0-9]*"`,
+	`SUBSYSTEM=="block", KERNEL=="loop[0-9]*"`,
+	`SUBSYSTEM=="misc", KERNEL=="loop-control"`,
+	`SUBSYSTEM=="misc", KERNEL=="device-mapper"`,
+	`SUBSYSTEM=="block", KERNEL=="dm-[0-9]*"`,
+}
+
+type microStackInterface struct {
+	commonInterface
+}
+
+var microStackSupportConnectedPlugKmod = []string{`vhost`, `vhost-net`, `vhost-scsi`, `vhost-vsock`, `pci-stub`, `vfio`, `nbd`, `dm-mod`, `dm-thin-pool`, `dm-snapshot`, `iscsi-tcp`, `target-core-mod`}
```

Comment:
minor: I would break this line into multiple lines (one per module), and maybe add a comment next to those modules whose name is not very explicative, like `target-core-mod` (which seems related to SCSI).

Reply:
Done.